packages/compass-import-export/src/modules/import.ts

```diff
@@ -439,25 +434,144 @@ export const cancelImport = () => {
-            path: name,
-            checked: true,
-            type: 'String', // will be detected by analyzeCSVFields
-          };
+        for (const unknownField of fields) {
+          // fields are both CSV fields (where you can assign a type and decide
+          // to include/exclude it) or placeholder ones.
+          // ie. for foo[0] we'll show a type dropdown (labelled "foo") which
+          // determines the types of all the elements in the array and for
+          // foo[1] we just leave a placeholder.
+          if ((unknownField as PlaceholderField).type === 'placeholder') {
+            continue;
+          }
+
+          const csvField = unknownField as FieldFromCSV;
+
+          let detected = result.fields[csvField.path].detected;
+          if (detected === 'undefined') {
+            // This is a bit of an edge case. If a column is always empty and
+            // "Ignore empty strings" is checked, we'll detect "undefined".
+            // We'll never actually insert undefined due to the checkbox, but
+            // undefined as a bson type is deprected so it might give the wrong
+            // impression. We could select any type in the selectbox, so the
+            // choice of making it null is arbitrary.
+            detected = 'null';
+          }
+
+          csvField.type = detected;
+
+          // Summarise why we picked number/mixed if that's what got detected
+          /*
+          if (['number', 'mixed'].includes(detected)) {
+            const parts = [];
+            for (const [type, summary] of Object.entries(
+              result.fields[csvField.path].types
+            )) {
+              parts.push(`${type} (${summary.count})`);
+            }
+            csvField.summary = parts.join(', ');
+          }
+          */
+          csvField.result = result.fields[csvField.path];
+        }
+
+        dispatch({
+          type: SET_PREVIEW,
+          fields,
+          values,
+        });
+
+        dispatch({
+          type: ANALYZE_FINISHED,
+          result,
         });
+      })
+      .catch((err) => {
+        log.error(
+          mongoLogId(1_001_000_180),
+          'Import',
+          'Failed to analyze CSV fields',
+          err
+        );
+      });
+
+    dispatch({
+      type: ANALYZE_FAILED,
+    });
+  };
+};
+
+const loadCSVPreviewDocs = (): ThunkAction<
+  void,
+  RootImportState,
+  void,
+  AnyAction
+> => {
+  return (
+    dispatch: ThunkDispatch<RootImportState, void, AnyAction>,
+    getState: () => RootImportState
+  ): void => {
+    const { fileName, delimiter, analyzeAbortController } =
+      getState().importData;
+
+    // if there's already an analyzeCSVFields in flight, abort that first
+    if (analyzeAbortController) {
+      analyzeAbortController.abort();
+      dispatch({
+        type: ANALYZE_CANCELLED,
+      });
+    }
+
+    const input = fs.createReadStream(fileName);
+
+    listCSVFields({ input, delimiter })
+      .then((result) => {
```

Comment:
Would `await` syntax here? Or is there a reason to use the promise .then syntax?

Reply:
I just copied the style from the other thunk actions in the file.

Reply:
My preference would be towards using `await`, I find it makes error handling and readability a bit better. In my current work I'm updating one of the other occurrences in this file.

Reply:
I'll change it, then ;)